### ISSUE TITLE
feature: keydb multi-master setup support in omhiredis

### ISSUE THREAD

**Issue by Girgitt** [rsyslog/rsyslog#5447]
### Expected behavior
omhiredis should be able to write to clustered keydb where nodes  configured as active-replica.

TL;DR - this can be done by adding config option (e.g. action.skipRedisRoleCheck="off") that will disable this check:  https://github.com/rsyslog/rsyslog/blob/master/contrib/omhiredis/omhiredis.c#L247 but I am not familiar with rsyslog's code base and requirements for PRs (tests etc.). Also I do not know the process of managing the configuration parameters. If there are no special requirements I can issue a PR for this feature once I am done with a project requiring this functionality. 
Probably similar change would is needed for imhiredis but I have not used it yet - just anticipate similar problems.  

### Actual behavior
Omhiredis initially writes to clustered keydb but when re-connection is required due to (tested examples, other reasons may exist):
1. destination node change due to failover carried out by keepalived by migrating virtuap ip
2. connection closed due to client inactivity (when keydb is configured to terminate inactive connections)

the omhiredis module never re-connects and stays in a retry loop due to unsupported keydb role which is "active-replica" instead of expected "master"

### Steps to reproduce
1. run keydb cluster on two nodes with the following settings
```{bash}
/usr/sbin/keydb-server          --protected-mode no          --active-replica yes          --slave-read-only no          --pidfile /var/run/keydb/node_1.pid          --appendfilename appendonlyfil-node_1.aof          --port 6379          --logfile /var/log/keydb/keydb_node_1.log          --save 900 1 300 10 60 10000          --slaveof remote_node_ip_address 6379          --bind vip_ip own_ip          --timeout 60          --dir /keydbdata/u0          --client-output-buffer-limit slave 836870912 836870912 0 
```
2. configure keepalived or manually migrate VIP between keydb nodes e.g. with shell script
To add ip address to an interface
```{bash}
#!/bin/bash
INTERFACE="${vrrp_interface}"
IP_ADDRESS="${vip_ip}"
SUBNET_MASK="24"  

CONNECTION_NAME=$(nmcli -g NAME,DEVICE connection show | awk -F: -v iface="${vrrp_interface}" '$2 == iface {print $1}')

if [ -z "$CONNECTION_NAME" ]; then
    echo "No connection found for interface ${vrrp_interface}. Exiting."
    exit 1
fi


nmcli con mod "$CONNECTION_NAME" +ipv4.addresses "$IP_ADDRESS/$SUBNET_MASK"

nmcli con mod "$CONNECTION_NAME" ipv4.method manual

nmcli con mod "$CONNECTION_NAME" connection.autoconnect yes

nmcli con down "$CONNECTION_NAME" && nmcli con up "$CONNECTION_NAME"

echo " >> IP address $IP_ADDRESS/$SUBNET_MASK has been added to $CONNECTION_NAME and activated."

```

To remove ip address to an interface
```{bash}
#!/bin/bash
INTERFACE="${vrrp_interface}"
IP_ADDRESS="${vip_ip}"
SUBNET_MASK="24"  

CONNECTION_NAME=$(nmcli -g NAME,DEVICE connection show | awk -F: -v iface="${vrrp_interface}" '$2 == iface {print $1}')

if [ -z "$CONNECTION_NAME" ]; then
    echo "No connection found for interface ${vrrp_interface}. Exiting."
    exit 1
fi


nmcli con mod "$CONNECTION_NAME" -ipv4.addresses "$IP_ADDRESS/$SUBNET_MASK"

nmcli con mod "$CONNECTION_NAME" ipv4.method manual

nmcli con mod "$CONNECTION_NAME" connection.autoconnect yes

nmcli con down "$CONNECTION_NAME" && nmcli con up "$CONNECTION_NAME"

echo " >> IP address $IP_ADDRESS/$SUBNET_MASK has been removed from connection $CONNECTION_NAME."

```

3. compile omhiredis for any recnet rsysog tag e.g. 8.2406.0
4. start rsyslog with conig
```{conf}
global(workDirectory="/var/lib/rsyslog")

$ActionResumeRetryCount -1     # Retry indefinitely
$ActionResumeInterval 1       # Retry every 10 seconds
$ActionQueueDiscardMark 9000  # Set discard mark
$RepeatedMsgReduction off      # Disable repeated message reduction


module(load="builtin:omfile" Template="RSYSLOG_TraditionalFileFormat")

#include(file="/etc/rsyslog.d/*.conf" mode="optional")


module(load="imuxsock" 	  # provides support for local system logging (e.g. via logger command)
       SysSock.Use="off") # Turn off message reception via local log socket;
			  # local messages are retrieved through imjournal now.
#module(load="imjournal" 	    # provides access to the systemd journal
#       StateFile="imjournal.state") # File to store the position in the journal


module(load="imtcp") # needs to be done just once
input(type="imtcp" port="5140")

module(load="imudp") # needs to be done just once
input(type="imudp" port="5140")


*.info;mail.none;authpriv.none;cron.none                /var/log/messages
authpriv.*                                              /var/log/secure
mail.*                                                  -/var/log/maillog
cron.*                                                  /var/log/cron
*.emerg                                                 :omusrmsg:*
uucp,news.crit                                          /var/log/spooler
local7.*                                                /var/log/boot.log

module(load="/opt/rsyslog/bin/omhiredis" log.debug="on")  # Load the hiredis module
$DebugLevel 2

# Define Redis server details
action(
   type="omhiredis"
   server="vip_of_keydb_cluster"      # Redis server IP
   serverport="6379"             # Redis server port
   key="rsyslog_stream_keydbTest"     # Name of the Redis stream to store logs
   template="RSYSLOG_TraditionalFileFormat"  # Format of log messages
   mode="stream"           # Specify that we're writing to a Redis stream
    
   queue.type="linkedlist"         # Enable a queue (in-memory, use 'disk' for disk-based)
   queue.size="10000"              # Queue up to 10,000 messages
   queue.saveonshutdown="on"       # Save the queue to disk on shutdown if disk queue
   action.resumeRetryCount="-1"    # Keep retrying indefinitely if Redis is down
   action.resumeInterval="1"       # Retry connection every 10 seconds
   action.reportSuspension="on"    # Log when action is suspended due to Redis being down
   action.reportSuspensionContinuation="on"  # Log when the action resumes
   stream.capacityLimit="100000"   # stream size to trib by redis

)

```

5. log some messages e.g. from commandline
```{bash}
# logger -n 127.0.0.1 -P 5140 "test4 zażółcić gęślą jaźń"
```

6. preform VIP failover or wait more than a minute, then log something again
7. review keydb content - logging will stop
8. open rsysog's log - you should see message "omhiredis: current connected node is not a master" repeated every time omhiredis tries to resume redis connection
9. forwarding to keydb never resumes on its own but does resume when rsyslog is restarted and works till next re-connection attempt. 

### Environment
- rsyslog version: 8.2406.0 (tested other tags as well. Issue is not related to version but design decisions in omhiredis)
- platform: Rocky 9.2 amd64
